index.js

```diff
@@ -188,7 +193,17 @@ module.exports = {
             const groupKey = lodashGet(msg, 'value.content.groupKey')
             const groupId = lodashGet(msg, 'value.content.recps[0]')
 
-            ssb.box2.addGroupInfo(groupId, { key: groupKey, root: groupRoot })
+            ssb.box2.getGroupKeyInfo(groupId, (err, info) => {
```

Comment:
These steps are asynchronous, but the pull.drain() above doesn't wait for these asynchronous steps to end. So it's possible we end up with a lot of these calls running in parallel. So we need to change this `pull.drain` to a `pull.asyncMap()` and then add an empty/dummy `pull.drain` below that. The asyncMap can also be a paramap, but asyncMap is okayish.

Reply:
done